app/models/student.rb

```diff
@@ -0,0 +1,9 @@
+class Student < ApplicationRecord
+
+  has_many :transactions, dependent: :destroy
+  has_many :challenges, through: :transactions
+  belongs_to :batch
+  has_many :chllenge_acceptances, dependent: :destroy
+  has_many :challenges, through: :chllenge_acceptances
```

Comment:
Also, there's something wrong here. A student should not be able to have two `has_many :challenges`. If we think about it, when we call `@student.challenges`, then which one will rails resolve to?

Reply:
Also, there appears to be a typo for the `:chllenge_acceptances` table? 😅 Lets fix that first